src/api/api-utils.ts

```diff
@@ -74,7 +77,15 @@ export const fetchApi = async (
   console.log(`${method} ${urlWithParams}`);
 
   try {
-    return await fetch(urlWithParams, options);
+    let response = await fetch(urlWithParams, options);
+
+    if (response.status === 403 && isAuthorized && !refreshed) {
+      refreshed = true;
+      response = await tryRefreshAndRetry(endpoint, method, body, queryParams);
+    } else {
+      refreshed = false;
+    }
+    return response;
```

Comment:
Troche tu jest ta logika pokrecona. Co jakby utworzyc nowa funkcje fetchApiWithRefresh, ktora przyjmowala by te same parametry i dzialala nastepujaco:
 - wywloaj fetchApi
 - zobacz reponse
 - odswiez token
 - jeszcze raz wywolaj fetchApi
 - zwroc response
Dodatkowo tu przydalaby sie chyba logika, ktora nie pozwolilaby zrobic dwoch refresh requestow. Zeby przez jakies race condition sie nie wylogowac przypadkiem jak nam zrevokuje tokeny :D.

Np jakis glupi if z `isRefreshing` i czekaniem az sie zmieni wartosc. Trzeba by poczytac jak to da sie zrobic.

Ale pozniej ogolnie dzialac na nowej funkcji wszedzie zamiast na `fetchApi` i powinno byc git?